CMakeLists.txt

```diff
@@ -170,6 +170,10 @@ if (BOOST_ROOT)
   set(Boost_NO_BOOST_CMAKE ON)
 endif (BOOST_ROOT)
 
+if (BOOST_STATIC)
+  set(Boost_USE_STATIC_LIBS ON)
```

Comment:
IMO that should be part of the toolchain files, i.e. see:
https://github.com/xdata-skylark/libskylark/blob/development/CMake/toolchains/BGQ-clang-essl.cmake#L16